demisto_sdk/commands/content_graph/strict_objects/base_strict_model.py

```diff
@@ -0,0 +1,179 @@
+)
+DEFAULT_DYNAMIC_MODEL = create_dynamic_model(
+    field_name="defaultValue",
+    type_=Optional[Any],
+    default=None,
+    include_without_suffix=True,
+)
+ID_DYNAMIC_MODEL = create_dynamic_model(
+    field_name="id",
+    type_=Optional[Any],
+    default=None,
+    include_without_suffix=True,
+)
+
+
+class CommonFields(BaseStrictModel):
+    id_: str = Field(..., alias="id")
+    version: int
+    id_xsoar: str = Field(None, alias="id:xsoar")
+    id_marketplacev2: str = Field(None, alias="id:marketplacev2")
+    id_xsoar_saas: str = Field(None, alias="id:xsoar_saas")
+    id_xsoar_on_prem: str = Field(None, alias="id:xsoar_on_prem")
+
+
+class Argument(BaseModel):
+    __base__ = (
+        BaseStrictModel,
+        NAME_DYNAMIC_MODEL,
+        REQUIRED_DYNAMIC_MODEL,
+        DESCRIPTION_DYNAMIC_MODEL,
+        DEPRECATED_DYNAMIC_MODEL,
+        DEFAULT_DYNAMIC_MODEL,
+    )
+    name: str
+    required: Optional[bool] = None
+    default: Optional[bool] = None
+    description: str
+    auto: Optional[Auto] = None
+    predefined: Optional[List[str]] = None
+    is_array: Optional[bool] = Field(None, alias="isArray")
+    secret: Optional[bool] = None
+    deprecated: Optional[bool] = None
+    type: Optional[str] = None
+    hidden: Optional[bool] = None
+
+
+class Output(BaseStrictModel):
+    content_path: Optional[str] = Field(None, alias="contentPath")
+    context_path: Optional[str] = Field(None, alias="contextPath")
+    description: str
+    type: Optional[str] = None
+
+
+class Important(BaseModel):
+    __base__ = (DESCRIPTION_DYNAMIC_MODEL,)
+    # not inheriting from StrictBaseModel since dynamic_models do
+    context_path: str = Field(..., alias="contextPath")
+    description: str
+    related: Optional[str] = None
+    description_xsoar: Optional[str] = Field(None, alias="contextPath")
+    description_marketplacev2: Optional[str] = Field(
+        None, alias="description:marketplacev2"
+    )
+    description_xpanse: Optional[str] = Field(None, alias="description:xpanse")
+    description_xsoar_saas: Optional[str] = Field(None, alias="description:xsoar_saas")
+    description_xsoar_on_prem: Optional[str] = Field(
+        None, alias="description:xsoar_on_prem"
+    )
+
+
+class ScriptType(StrEnum):
+    PWSH = TYPE_PWSH
+    PYTHON = TYPE_PYTHON
+    JS = TYPE_JS
+
+
+class StructureError(BaseStrictModel):
+    field_name: Optional[tuple] = Field(
+        None, alias="loc"
+    )  # the api returns here tuple, not str for this key
+    error_message: Optional[str] = Field(None, alias="msg")
+    error_type: Optional[str] = Field(None, alias="type")
+    ctx: Optional[dict] = None
+
+
+class BaseIntegrationScript(BaseStrictModel):
+    __base__ = (NAME_DYNAMIC_MODEL, DEPRECATED_DYNAMIC_MODEL)
+    name: str
+    deprecated: Optional[bool] = None
+    from_version: Optional[str] = Field(None, alias="fromversion")
+    to_version: Optional[str] = Field(None, alias="toversion")
```

Comment:
We'll have a lot of `fromversion`/`toversion`, worth moving them to a different object
(it's formatted `fromversion` in ymls, `fromVersion` in jsons) 

`name`, `system` as well, try to find the smallest common denominator to divide them in a pretty way